server/plugin.go

```diff
@@ -73,15 +71,14 @@ type Plugin struct {
 
 	WebhookHandler webhook.Webhook
 	GitlabClient   gitlab.Gitlab
-	// gitlabPermalinkRegex is used to parse gitlab permalinks in post messages.
-	gitlabPermalinkRegex *regexp.Regexp
 }
 
+// gitlabPermalinkRegex is used to parse gitlab permalinks in post messages.
+var gitlabPermalinkRegex = regexp.MustCompile(`https?://(?P<haswww>www\.)?gitlab\.com/(?P<user>[\w/.?-]+)/(?P<repo>[\w-]+)/-/blob/(?P<commit>\w+)/(?P<path>[\w-/.]+)#(?P<line>[\w-]+)?`)
```

Comment:
any reason why we cannot make this a constant

Reply:
I was facing a compilation error while making it constant because when we are trying to assign to a constant that has a type that can't be constant (like for example, Regexp). Only basic types likes int, string, etc. can be constant.